### PR TITLE
1898 added subject page and made subject filtering bette

### LINKED ISSUE #1898: Filtering preprints by sub-discipline
**Is your feature request related to a problem? Please describe.**
On the old EarthArXiv website, we used to be able to browse preprints and filter them by discipline/tag, e.g. "Geophysics and Seismology". This was useful for quickly browsing new preprints most relevant to the user.

**Describe the solution you'd like**
Currently, a workaround is to search for the "geophysics and seismology" tag, which sort of works, but the search results cannot be sorted by date, so it is not easy to quickly browse the newest preprints which are most appropriate to you. The sub-discipline tickbox options on the old EarthArXiv website worked well, I think.

### PR DESCRIPTION
**Pull request by ajrbyers**
- Adds a subject list page
- Updates subject filter page
Closes #1898

### CODE REVIEW

File: src/repository/views.py [hunk 1]
```diff
@@ -274,7 +294,7 @@ def repository_search(request, search_term=None):
         )]
 
         preprints = list(set(list(preprint_search) + preprints_from_author))
-
+        preprints.sort(key=lambda x: x.date_published, reverse=True)
```
Comment (mauromsl) [inline src/repository/views.py:297]: Worth using `operator.attrgetter` here to squeeze some nanoseconds out of the sorting.

File: src/themes/material/templates/repository/elements/subject_tree.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{% for subject in subjects %}
+    <li><a href="{% url 'preprints_list_subject' subject.slug %}">{{ subject }}</a></li>
```
Comment (mauromsl) [inline src/themes/material/templates/repository/elements/subject_tree.html:2]: So it looks like the view routed from this URL predates this PR. I had a look at it and I don't think it does what it is supposed to do:
https://github.com/BirkbeckCTP/janeway/blob/d89a6114b0ed280b63f210316937ec0dcd989ea8/src/repository/views.py#L199

It fetches the subject but does not apply any filters on the queryset.

While we are at it, is it worth renaming this to repository_list_subject to be consistent with the other views?

**Comment by shicks-seismo** [comment]
Nice - speedy work - thanks!

File: src/themes/material/templates/repository/elements/subject_tree.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{% for subject in subjects %}
```
Comment (mauromsl) [inline src/themes/material/templates/repository/elements/subject_tree.html:1]: Just a nice to have: we could add a count of articles on each subject

**Review by mauromsl — CHANGES_REQUESTED** [review]

File: src/themes/material/templates/repository/elements/subject_tree.html (hunk 2 above)
Comment (ajrbyers) [reply src/themes/material/templates/repository/elements/subject_tree.html:2]: It grabs the subject and then fetches its preprint_set see:

https://github.com/BirkbeckCTP/janeway/blob/d89a6114b0ed280b63f210316937ec0dcd989ea8/src/repository/views.py#L201

Comment (mauromsl) [reply src/themes/material/templates/repository/elements/subject_tree.html:2]: :+1:

**Pull request merged**